src/moai-android-adcolony/host.cpp

```diff
@@ -1,6 +1,16 @@
 // Copyright (c) 2010-2011 Zipline Games, Inc. All Rights Reserved.
 // http://getmoai.com
 
+// From moai-box2d's pch.h
+#include <moai_config.h>
+#include <zl-util/headers.h>
+#include <moai-core/headers.h>
+#include <moai-sim/headers.h>
```

Comment:
Without the changes, build-android gives:

D:\MOAI-Stuff\moai-dev3\src\moai-android-adcolony\host.cpp: In function 'void AKUAndroidAdColonyContextInitialize()':

D:\MOAI-Stuff\moai-dev3\src\moai-android-adcolony\host.cpp:29:23: error: 'MOAIAdColonyAndroid' was not declared in this scope

D:\MOAI-Stuff\moai-dev3\src\moai-android-adcolony\host.cpp:29:43: error: 'REGISTER_LUA_CLASS' was not declared in this scope

make[2]: *** [libmoai/moai-android/moai-android-adcolony/CMakeFiles/moai-android-adcolony.dir/D_/MOAI-Stuff/moai-dev3/src/moai-android-adcolony/host.cpp.o] Error 1

make[1]: *** [libmoai/moai-android/moai-android-adcolony/CMakeFiles/moai-android-adcolony.dir/all] Error 2

The issue actually is that nothing has pulled in 
<host-modules/aku_modules_android_config.h> leaving 
AKU_WITH_ANDROID_ADCOLONY undefined.  MOAIAdColonyAndroid.h therefore 
doesn't define the class, but MOAIAdColonyAndroid.cpp attempts to 
implement the undefined class.

It turns out that simply adding 
<host-modules/aku_modules_android_config.h> to MOAIAdColonyAndroid.h to 
get the definition of AKU_WITH_ANDROID_ADCOLONY works.  However, if that 
is defined to 0, then the original issue comes back.

The class implementation in MOAIAdColonyAndroid.cpp should be wrapped in 
#if AKU_WITH_ANDROID_ADCOLONY to finish the configuration of the host 
component.

This needs to be done to each of the android host components.  I'm 
willing to do this and remove the then unnecessary headers from the 
corresponding host.cpp files if this is deemed as the proper solution.

Lynn (D)




On 4/1/2015 3:09 AM, halfnelson wrote:
>
> In src/moai-android-adcolony/host.cpp 
> <https://github.com/moai/moai-dev/pull/1102#discussion_r27548794>:
>
> > @@ -1,6 +1,16 @@
> >  // Copyright (c) 2010-2011 Zipline Games, Inc. All Rights Reserved.
> >  // http://getmoai.com
> >
> > +// From moai-box2d's pch.h
> > +#include <moai_config.h>
> > +#include <zl-util/headers.h>
> > +#include <moai-core/headers.h>
> > +#include <moai-sim/headers.h>
>
> these were all compiling fine for me on android? why drag in all these 
> headers?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/moai/moai-dev/pull/1102/files#r27548794>.
>